src/plugins/console/public/lib/utils/index.ts

```diff
@@ -108,3 +113,55 @@ export const getResponseWithMostSevereStatusCode = (requestData: RequestResult[]
       .pop();
   }
 };
+
+export const replaceVariables = (
+  requests: RequestArgs['requests'],
+  variables: DevToolsVariable[]
+) => {
+  // Default variables will be used in the default request to demonstrate variables in action
+  variables = [...DEFAULT_VARIABLES, ...variables];
+  const urlRegex = /(\${\w+})/g;
+  const bodyRegex = /("\${\w+}")/g;
+  return requests.map((req) => {
+    if (urlRegex.test(req.url)) {
+      req.url = req.url.replaceAll(urlRegex, (match) => {
+        // Sanitize variable name
+        const key = match.replace('${', '').replace('}', '');
+        const variable = variables.find(({ name }) => name === key);
+
+        return variable?.value ?? match;
+      });
+    }
+
+    if (req.data.length) {
+      if (bodyRegex.test(req.data[0])) {
+        const data = req.data[0].replaceAll(bodyRegex, (match) => {
+          // Sanitize variable name
+          const key = match.replace('"${', '').replace('}"', '');
+          const variable = variables.find(({ name }) => name === key);
+
+          if (variable) {
+            const { value } = variable;
+            if (
+              // Stringify if value is an object, array, number or boolean, this is necessary in order to send a successful request to es
```

Comment:
I think this comment should state "Stringify if value is **not** an...", right? At least it looks like that's what the logic is doing.

Actually, I'm a little confused. When we say "the value is an object" in this context, are we talking about an object represented as a string, like this?

```js
// This is considered an "object" in this context
'{"a": "b"}'
```

Reply:
After playing with this a bit, I think I understand the point here is to stringify all values being sent to ES.

Assuming this is correct, I'd suggest making this logic a little more verbose so it's easier to follow. WDYT?

```js
if (variable) {
  // All values must be stringified to send a successful request to ES.
  const { value } = variable;

  const isStringifiedObject = value.startsWith('{') && value.endsWith('}');
  if (isStringifiedObject) {
    return value;
  }

  const isStringifiedNumber = !isNaN(parseFloat(value));
  if (isStringifiedNumber) {
    return value;
  }

  const isStringifiedArray = value.startsWith('[') && value.endsWith(']');
  if (isStringifiedArray) {
    return value;
  }

  const isStringifiedBool = value === 'true' || value === 'false';
  if (isStringifiedBool) {
    return value;
  }

  // At this point the value must be an unstringified string, so we have to stringify it.
  // Example: 'stringValue' -> '"stringValue"'
  return JSON.stringify(value);
}
```

Reply:
Awesome!